### ISSUE TITLE
Rule: Variables and functions should not be redeclared #16 false positive

### ISSUE THREAD

**Issue by janv8000** [SonarSource/SonarTS#899]
**I want to report a bug.**

**SonarTS version:** SonarCloud

**Rule key:** typescript:S2814

**Reproducer** 
```
declare namespace Company.Product.WebApi.Features.Resource.GetResource.GetResource {
    interface Request {
        type: ResourceType;
    }
    interface Response {
        result: number;
    }
}
```
Rule message: 'GetResource' is already defined.

**Expected behavior**

Should not warn.

These files get generated by TypeLite
